File: Foldable Binary Tree.java
boolean mirror(Node root1, Node root2) {
        if(root1 == null && root2 == null) {
            return true;
        }
        if(root1 == null || root2 == null) {
            return false;
        }
        
        return mirror(root1.left, root2.right) && mirror(root1.right, root2.left);
    }
    
    boolean IsFoldable(Node node) { 
		if(node == null) return true;

		return mirror(node.left, node.right);
	}

File: Foldable Binary Tree.cpp
void inorder(vector<int> &v, Node* node, int level){
    if(node==NULL)  return;
    inorder(v, node->left, level+1);
    v.push_back(level);
    inorder(v, node->right, level+1);
}

bool ispalin(vector<int> &v){
    int n = v.size();
    for(int i=0;i<n/2;i++)
        if(v[i]!=v[n-i-1])
            return false;
    return true;
}

bool IsFoldable(Node* root)
{
    vector<int> v;
    inorder(v, root, 0);
    return ispalin(v);
}
